src/Type/Doctrine/Query/QueryResultTypeWalker.php

```diff
@@ -1108,7 +1130,21 @@ public function walkLiteral($literal)
 				if (floatval(intval($value)) === floatval($value)) {
 					$type = new ConstantIntegerType((int) $value);
 				} else {
-					$type = new ConstantFloatType((float) $value);
+
+					if ($this->em->getConnection()->getDatabasePlatform() instanceof AbstractMySQLPlatform) {
+
+						// both PDO_mysql and mysqli hydrates 123.4 literals as string no matter the configuration (e.g. PDO::ATTR_STRINGIFY_FETCHES being false) and PHP version
+						// the only way to force float is to use 123.4e0 scientific notation
+						// https://dev.mysql.com/doc/refman/8.0/en/number-literals.html
```

Comment:
Maybe this should mention that it's because 123.4 is a decimal literal, while 123.4e0 is a float literal.

Reply:
Added